src/gui/guiFormSpecMenu.cpp

```diff
@@ -2459,6 +2460,31 @@ bool GUIFormSpecMenu::parseStyle(parserData *data, const std::string &element, b
 
 	StyleSpec spec;
 
+	// Parse state information, if it exists
+	size_t state_pos = parts[0].find(':');
+	if (state_pos != std::string::npos) {
+		selector = selector.substr(0, state_pos);
+		std::string state_str = parts[0].substr(state_pos + 1);
+
+		if (state_str.empty()) {
+			errorstream << "Invalid style element (Invalid state): '" << element
+				<< "'" << std::endl;
+		} else {
+			std::vector<std::string> states = split(state_str, '+');
+			for (std::string &state : states) {
+				StyleSpec::State converted = StyleSpec::getStateByName(state);
+				if (converted == StyleSpec::STATE_INVALID) {
+					infostream << "Unknown style state " << state <<
+							" in element '" << element << "'" << std::endl;
+					return false;
```

Comment:
The `errorstream` log above doesn't return but this `infostream` does. I'd suggest to change this to `continue;`.

Reply:
This should not be `continue`. The end-result of that would be breaking styling on old clients if new states were added, instead of correctly ignoring the states (For more details, see the short discussion with @rubenwardy earlier in the thread). You're right that the lack of return on the error is inconsistent though, I'll add one there.